### DISCUSSION TITLE
TransformManager deparenting

### DISCUSSION THREAD

**Discussion by ThomasGorisse** [google/filament#3748] [Q&A]
Hi,

From what I understand, `TransformManager` is able to manage custom Entity parenting.

- Can you confirm that it is a complete hierarchical handler = If I change the transform of a parent the transformation will be applied to all the childs hierarchies ?

- If I want to deparent an EntityInstance, What is the right way to call the 
`public void setParent(@EntityInstance int i, @EntityInstance int newParent)` ?
    - Passing newParent = EntityInstance.NULL
    - Passing newParent = scene.getRoot() (which we don't have access to)

- Can you also confirm that I well understood the fact that an "EntityInstance" without parent/with scene parent will return the same value from getTransform() and getWorldTransform() ?

Thanks

**Comment by romainguy** [comment]
The answer to questions 1 and 3 is yes. For question 2, I don't remember, @pixelflinger would know.

**Comment by ThomasGorisse** [comment]
One more question:
4. If call setParent on an EntityInstance wich already has a custom transform will the actual child transform be applied based on the parent transform or rested to parent transform ?

**Reply by romainguy** [reply]
An entity always has its own transform and, optionally, a parent transform. The final transform used to render is the composition of both transforms. If you reparent, the child's transform will be composed with the new parent's transform.
